### PR TITLE
lint

### PR DESCRIPTION
**Pull request by mscdex**

### CODE REVIEW

**Comment by bnoordhuis** [comment]
We don't do lint here at joyent/libuv but your PR did catch a syntax error that I've fixed in 3a8c398.

@piscisaureus The missing free() in src/win/fs.c looks like a genuine bug to me.

**Comment by mscdex** [comment]
So unused/useless code is ok? :confused:

**Comment by piscisaureus** [comment]
I'm still reviewing but so far I've seen good improvements. Hang on.

File: src/win/poll.c [hunk 1]
```diff
@@ -319,11 +319,7 @@ static DWORD WINAPI uv__slow_poll_thread_proc(void* arg) {
   assert(handle->type == UV_POLL);
   assert(req->type == UV_POLL_REQ);
 
-  if (req == &handle->poll_req_1) {
-    events = handle->submitted_events_1;
-  } else if (req == &handle->poll_req_2) {
-    events = handle->submitted_events_2;
-  } else {
+  if (req != &handle->poll_req_1 && req != &handle->poll_req_2) {
```
Comment (piscisaureus) [inline src/win/poll.c]: Good catch on the superfluous `events` var and assignment, but I'm sure that the construct `if (!condition) assert(0)` wasn't in the style guide either. I'd just drop the check altogether.

**Comment by bnoordhuis** [comment]
> So unused/useless code is ok?

If it serves a purpose, like being documentation for the next programmer. As an example:

```
-  /* We've got a copy, don't touch the original any more. */
-  name = NULL;
```

That tells me or whoever comes after me that name should not be used after that point.

File: src/win/tty.c [hunk 2]
```diff
@@ -96,10 +96,9 @@ void uv_console_init() {
 
 
 int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, uv_file fd, int readable) {
-  HANDLE handle = INVALID_HANDLE_VALUE;
+  HANDLE handle = (HANDLE) _get_osfhandle(fd);
```
Comment (piscisaureus) [inline src/win/tty.c]: Disagree on moving the assignment to the declaration. Just drop the initialization with INITIAL_HANDLE_VALUE.

File: src/win/util.c [hunk 3]
```diff
@@ -579,7 +579,7 @@ uv_err_t uv_uptime(double* uptime) {
 
 
 uv_err_t uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
-  SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION* sppi;
+  SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION* sppi = NULL;
```
Comment (piscisaureus) [inline src/win/util.c:582]: why?

File: src/inet.c [hunk 4]
```diff
@@ -61,7 +61,7 @@ static uv_err_t inet_ntop4(const unsigned char *src, char *dst, size_t size) {
 #else
   l = _snprintf(tmp, sizeof(tmp), fmt, src[0], src[1], src[2], src[3]);
 #endif
-  if (l <= 0 || l >= size) {
+  if (l == 0 || l >= size) {
```
Comment (piscisaureus) [inline src/inet.c]: disagree. `_snprintf` returns `int` on windows.

**Comment by bnoordhuis** [comment]
Also, I hate getting sidetracked by lint commits when I'm doing code archeology.

File: src/win/tty.c (hunk 2 above)
Comment (mscdex) [inline src/win/tty.c]: fine by me

File: src/unix/getaddrinfo.c [hunk 5]
```diff
@@ -88,7 +88,7 @@ static void uv__getaddrinfo_done(struct uv__work* w, int status) {
   else if (req->retcode == EAI_NODATA)
     uv__set_sys_error(req->loop, ENOENT);
 #elif defined(__sun)
-  if (req->retcode == EAI_MEMORY && hostlen >= MAXHOSTNAMELEN) {
```
Comment (piscisaureus) [inline src/unix/getaddrinfo.c:91]: How did this ever work? Nobody ever compiled on __sun ?

File: src/win/util.c (hunk 3 above)
Comment (mscdex) [inline src/win/util.c:582]: Due to a possible early bailout on [this line](https://github.com/joyent/libuv/blob/master/src/win/util.c#L603)? There is an `if (sppi)` check right after the :out label and sppi would not necessarily be initialized to null?

File: src/inet.c (hunk 4 above)
Comment (mscdex) [inline src/inet.c]: isn't size_t unsigned?

File: src/win/poll.c (hunk 1 above)
Comment (mscdex) [inline src/win/poll.c]: fine by me

File: src/win/util.c (hunk 3 above)
Comment (piscisaureus) [inline src/win/util.c:582]: Good catch!

**Comment by mscdex** [comment]
> > So unused/useless code is ok?
> 
> If it serves a purpose, like being documentation for the next programmer. As an example:
> 
> ```
> -  /* We've got a copy, don't touch the original any more. */
> -  name = NULL;
> ```
> 
> That tells me or whoever comes after me that name should not be used after that point.

Ok, so leaving just the comment in this particular case would be fine?

**Comment by bnoordhuis** [comment]
No. We're poisoning the variable. If someone does inadvertently dereference the pointer, s/he will find out soon enough because it'll segfault.

**Comment by mscdex** [comment]
Updated commit as per suggestions.

File: src/inet.c (hunk 4 above)
Comment (mscdex) [inline src/inet.c]: i've reverted this and instead changed `l` to be `int` instead of `size_t`.

**Comment by piscisaureus** [comment]
Thanks Brian, all the issues you found are addressed in the following commits. Some "useless" code removal wasn't landed because it serves a purpose.

https://github.com/joyent/libuv/commit/4d4f1496d9b600dfd3e4c4f0040d7b4a0d41043c
https://github.com/joyent/libuv/commit/9021dbcd8d1dc3b1c6cca82f46d5e774088d4ae6
https://github.com/joyent/libuv/commit/a9bce29f0c98187fcf09d1ba9dcd4eb5548fb30e
https://github.com/joyent/libuv/commit/7570a35b70a33258e4c36e080e900f30c3213697
https://github.com/joyent/libuv/commit/603915dd0081ec2f34d612291a1aeb909f6f2b76

**Pull request closed without merging**